### DISCUSSION TITLE
Concatenation of select boxes

### DISCUSSION THREAD

**Discussion by AE-SAY-WAY** [mdelobelle/metadatamenu#521] [Ideas]
Fields that show up only if a specific property is selected.

For example,

The only way that other property selection menus come up, if a prerequisite property was selected. Sort of like a conditional property.

**Comment by AE-SAY-WAY** [comment]
For more context,
I mean if a specific value OF a property is selected. Another field would pop up depending on the value selected. Exactly as the name suggests a conditional property type.

**Comment by AE-SAY-WAY** [comment]
Maybe we can have conditional property in which the values that appear contain the concatenation of other select boxes values'.

Name: John (select box)
Lastname: Mcdun (select box)

Name: Jason (select box)
Lastname: matthews (select box)

John Mcdun (a product of the first select box)
Jason Matthews (a product of the second select box)

And then maybe we can give these special combinations, their own alias.

so you would select John for one property, you would select Mcdun as another property, and then a third property would take those 2 and combine them to get John Mcdun and then you can use the alias function to rename it.

**Reply by mdelobelle** [reply]
here is something i've recommended on discord:

---

ok I have something quite complex but working:

Given this folder structure:
```
Character
|--Families
   |--Character A
   |--Character B
|--Gods
   |-- Character Goddess of creation
|--Main Character
   |--Character Mac Zafier
|--Side Character
   |--Character Fl Liara

```

I want to use them in a `Scene` class file

I define a fileClass `Scene` with two fields:
**first field:**
`Character Type` 
of type `Select` 
with the option `Values returned from a dataview query` 
and the query being:
```
Object.values(app.vault.adapter.files)
    .filter(item => item.type === "folder" && 
        item.realpath.startsWith("Character/")
    ).map(item => item.realpath)
```
in fact it's a pure javascript query returning all the sub-folders of `Character`

**second field:**
`Character`
of type `Select` 
with the option `Values returned from a dataview query` 
the query being (more complicated but will explain)
```
(() => {
    const path = current["Character Type"]
    if(path){
        return Object.values(app.vault.adapter.files)
            .filter(item => item.type==="file" &&
                item.realpath.startsWith(path)
            ).map(item => item.realpath
                .replace(path+"/", "")
                .replace(".md","")
            )
    }
     return []
})()
```
this is a self evaluating function, it gets the value of `Character Type`and use it to return a (cleaned) list of file names at the corresponding folder path

here are some screenshots:
![image](https://github.com/mdelobelle/metadatamenu/assets/1233240/c5fa0ad1-52a4-4bdb-aa38-e43907e840d2)
![image](https://github.com/mdelobelle/metadatamenu/assets/1233240/99e82fb2-2d5a-4e4d-83be-a40c4a4c737f)
![image](https://github.com/mdelobelle/metadatamenu/assets/1233240/32db17d9-1b4d-490e-8090-6ceeace573cc)
